### PR TITLE
DOC: Update jnp.prod docs

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -309,11 +300,78 @@ def _reduce_prod(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None
...
+  Returns:
+    An array of the product along the given axis.
+
+  See also:
+    - :func:`jax.numpy.sum`: Compute the sum of array elements over a given axis.
+    - :func:`jax.numpy.max`: Compute the maximum of array elements over given axis.
+    - :func:`jax.numpy.min`: Compute the minimum of array elements over given axis.
+
+  Examples:
+
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:337]: Remove blank line

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -309,11 +300,78 @@ def _reduce_prod(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None
...
+           [ 6]], dtype=int32)
+
+    To include only specific elements in the sum, you can use a``where``.
+
+    >>> where=jnp.array([[1, 0, 1, 0],
+    ...                  [0, 0, 1, 1],
+    ...                  [1, 1, 1, 0]], dtype=bool)
+    >>> jnp.prod(x, axis=1, keepdims=True, where=where)
+    Array([[4],
+          [3],
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:365]: Missing space on lines 365-366 to align the output

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -309,11 +300,78 @@ def _reduce_prod(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None
...
+          [6]], dtype=int32)
+    >>> where = jnp.array([[False],
+    ...                    [False],
+    ...                    [False]])
+    >>> jnp.prod(x, axis=1, keepdims=True, where=where)
+    Array([[1],
+          [1],
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:372]: Missing space

**Review by jakevdp — COMMENTED** [review]
Looks good! Just a couple small changes.

Sorry to nitpick about formatting, but we're about to land #22037 which will make docstring formatting rules more strict.

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Removed.

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Included the space.

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Included the space and squashed into single commit

**Review by jakevdp — APPROVED** [review]
Thanks!

**Comment by jakevdp** [comment]
This PR and #22105 both failed to merge because they touch the same file as #22102. Can you rebase against the current main branch?

**Comment by jakevdp** [comment]
Sorry, I meant to rebase against the updated main branch, not merge the updated main branch.

Can you squash the commits please?

**Comment by rajasekharporeddy** [comment]
Squashed the commits.

**Pull request merged**